packages/runtime/test/src/interpreter/pooling-inner-transactions-test.ts

```diff
@@ -0,0 +1,375 @@
+        interpreter.runtime.ctx.state.txReceipts.set(TXN_OBJ.txID, {
+            txn: TXN_OBJ,
+            txID: TXN_OBJ.txID,
+        });
+    };
+
+    const setUpInterpreter = (
+        tealVersion: number,
+        appBalance: number = ALGORAND_ACCOUNT_MIN_BALANCE
+    ): void => {
+        // setup 1st account (to be used as sender)
+        elonAcc = new AccountStore(0, elonMuskAccount); // setup test account
+        setDummyAccInfo(elonAcc);
+
+        // setup 2nd account
+        johnAcc = new AccountStore(0, johnAccount);
+
+        // setup 3rd account
+        bobAccount = new AccountStore(1000000, bobAcc);
+
+        // setup application account
+        appAccAddr = getApplicationAddress(TXN_OBJ.apid);
+        applicationAccount = new AccountStore(appBalance, {
+            addr: appAccAddr,
+            sk: new Uint8Array(0),
+        });
+
+        // setup foreign application account
+        foreignAppAccAddr = getApplicationAddress(TXN_OBJ.apfa[1]);
+        foreignApplicationAccount = new AccountStore(appBalance, {
+            addr: foreignAppAccAddr,
+            sk: new Uint8Array(0),
+        });
+
+        interpreter = new Interpreter();
+        interpreter.runtime = new Runtime([elonAcc, johnAcc, bobAccount,
+            applicationAccount, foreignApplicationAccount]);
+        interpreter.tealVersion = tealVersion;
+        reset();
+    };
+
+    const executeTEAL = (tealCode: string, onComplete = TxOnComplete.NoOp): void => {
+        // reset interpreter
+        reset();
+        interpreter.runtime.ctx.tx.apan = Number(onComplete);
+        interpreter.execute(tealCode, ExecutionMode.APPLICATION, interpreter.runtime, 0);
+    };
+
+    describe("Pooling fee tests in group txns", function () {
+        this.beforeEach(function () {
+            setUpInterpreter(7, balance);
+        });
+
+        it("Should succeed: when unfunded account opts-in in a group txn", function () {
+            setUpInterpreter(7, 0);
+            const elonAcc = interpreter.runtime.ctx.state.accounts.get(elonAddr);
+            TXN_OBJ.apat.push(Buffer.from(decodeAddress(applicationAccount.address).publicKey));
+            TXN_OBJ.apat.push(Buffer.from(decodeAddress(bobAccount.address).publicKey));
+
+            if (elonAcc) {
+                elonAcc.amount = 1000000n;
+                const assetID = interpreter.runtime.ctx.deployASADef(
+                    "test-asa",
+                    { total: 10, decimals: 0, unitName: "TASA" },
+                    elonAddr,
+                    { creator: { ...elonAcc.account, name: "elon" } }
+                ).assetIndex;
+
+                bobAccount.account.rekeyTo(applicationAccount.address);
+
+                TXN_OBJ.apas = [assetID];
+
+                assert.equal(interpreter.runtime.ctx.state.accounts
+                    .get(johnAcc.address)?.balance()
+                    , 0n);
+
+                const prog = `
+                itxn_begin
```

Comment:
we have different formatting for the programs across the test lets fix it

Reply:
Done